nanoFirmwareFlasher/FirmwarePackage.cs

```diff
@@ -64,23 +63,75 @@ internal abstract class FirmwarePackage : IDisposable
             _preview = preview;
         }
 
-        public static List<CloudSmithPackageDetail> GetBoardList(bool communityTargets, bool preview, string filter, VerbosityLevel verbosity)
+        /// <summary>
+        /// Get a list of all available targets from Cloudsmith repository.
+        /// </summary>
+        /// <param name="communityTargets"><see langword="true"/> to list community targets.<see langword="false"/> to list reference targets.</param>
+        /// <param name="preview">Option for preview version.</param>
+        /// <param name="platform">Platform code to use on search.</param>
+        /// <param name="verbosity">VerbosityLevel to use when outputting progress and error messages.</param>
+        /// <returns>List of <see cref="CloudSmithPackageDetail"/> with details on target firmware packages.</returns>
+        public static List<CloudSmithPackageDetail> GetTargetList(
+            bool communityTargets,
+            bool preview,
+            SupportedPlatform? platform,
+            VerbosityLevel verbosity)
         {
-            string repoName = communityTargets ? _communityTargetsRepo : preview ? _refTargetsDevRepo : _refTargetsStableRepo;
-            string requestUri = $"{_cloudsmithPackages}/{repoName}/?query={filter}";
-            List<CloudSmithPackageDetail> boardNames = new List<CloudSmithPackageDetail>();
+            string queryFilter = string.Empty;
 
-            if (verbosity >= VerbosityLevel.Normal)
+            // build query filter according to platform using package name
+            if (platform != null)
             {
-                Console.ForegroundColor = ConsoleColor.White;
-                Console.Write($"Trying to find list of boards in {(preview ? "development" : "stable")} repository");
-                if( string.IsNullOrEmpty(filter))
+                List<string> query = new(); 
+
+                switch(platform)
                 {
-                    Console.Write(" without filter");
+                    case SupportedPlatform.esp32:
+                        query.Add("ESP");
+                        query.Add("M5");
+                        query.Add("Pyb");
+                        query.Add("FEATHER");
```

Comment:
missing lilygo

Reply:
That will be picked up by the ESP filter, the target name is `ESP32_LILYGO`.